crates/runtime/src/utils.rs

```diff
@@ -0,0 +1,30 @@
+use cairo_felt::Felt252;
+use num_traits::cast::ToPrimitive;
+use num_traits::identities::One;
+
+pub fn read_felt_and_increase_idx(buffer: &[Felt252], idx: &mut usize) -> Felt252 {
```

Comment:
I would stick with the previous names, the fact that idx changes is signalised by the signature

Reply:
I would create `Reader` struct which holds an index and buffer instead, would hide this implementation detail and be more concise. (Like Buffer in cario but not relying on pointers etc).

Reply:
We could go one step further and pass it to the function instead of `buffer: &[Felt252]`

Reply:
Also those functions problably shouldn't panic when parsing fails, input data could be anything as user can put anything as cheatcode argument.

Reply:
#1469

Reply:
the only place where it may panic is [here](https://github.com/foundry-rs/starknet-foundry/pull/1440/files#diff-4658fbded6e024a7e21c213e4b278f37d12083f980631a8b91d6f9a09f54685cR21), which is OK, since it indicates that this is not a vector

Reply:
I'll improve error msg

Reply:
@abulenok they were prob thinking of places like this too
https://github.com/foundry-rs/starknet-foundry/blob/e9daf1227c6589a4fc0c26aa22851a4e78a58c55/crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs#L549

Reply:
@piotmag769 this actually is probably out of scope of this work (as you mentioned), could we create a separate issue for stuff like this?